components/selection/selection-action.js

```diff
@@ -47,55 +43,47 @@ class Action extends LocalizeCoreElement(SelectionSubscriberMixin(ButtonMixin(Rt
 		`;
 	}
 
-	get selectionInfo() {
-		return super.selectionInfo;
-	}
-
-	set selectionInfo(value) {
-		super.selectionInfo = value;
-		this.nonInteractive = (this.requiresSelection && this.selectionInfo.state === SelectionInfo.states.none);
+	constructor() {
+		super();
+		this._selectionInfo = new SelectionInfoController(this);
 	}
 
 	connectedCallback() {
 		super.connectedCallback();
 		this.addEventListener('d2l-button-ghost-click', this._handleActionClick);
+		this._selectionInfo.hostConnected(); // remove with Lit 2
 	}
 
 	disconnectedCallback() {
 		super.disconnectedCallback();
 		this.removeEventListener('d2l-button-ghost-click', this._handleActionClick);
+		this._selectionInfo.hostDisconnected(); // remove with Lit 2
 	}
 
 	render() {
+		const nonInteractive = (this.requiresSelection && this._selectionInfo.state === SelectionInfo.states.none);
```

Comment:
Moved this out to a computed value since it didn't seem to need to be a public property?